lms/djangoapps/course_api/blocks/transformers/extra_fields.py

```diff
@@ -0,0 +1,71 @@
+"""
+Extra Fields Transformer
+"""
+import re
+
+from django.conf import settings
+
+from openedx.core.djangoapps.content.block_structure.transformer import BlockStructureTransformer
+
+
+class ExtraFieldsTransformer(BlockStructureTransformer):
+    """
+    Shows ExtraFields specified in COURSE_BLOCKS_API_EXTRA_FIELDS LMS Django Settings variable
+    """
+    WRITE_VERSION = 1
+    READ_VERSION = 1
+    EXTRA_FIELDS_DATA = 'extra_fields'
+
+    @classmethod
+    def name(cls):
+        return "blocks_api:extra_fields"
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Collects any information that's necessary to execute this transformer's
+        transform method.
+        """
+        requested_field_names = cls.get_requested_extra_fields()
+
+        block_structure.request_xblock_fields('category', *requested_field_names)
+
+    @classmethod
+    def get_requested_extra_fields(cls):
+        """
+        Returns the names of the requested extra fields
+        """
+        if settings.FEATURES.get('COURSE_BLOCKS_API_EXTRA_FIELDS') is None:
+            return []
+
+        return [field_name for block_type, field_name in settings.FEATURES.get('COURSE_BLOCKS_API_EXTRA_FIELDS')]
+
+    def transform(self, usage_info, block_structure):
+        """
+        Mutates block_structure based on the given usage_info.
+        """
+        if len(self.get_requested_extra_fields()) == 0:
+            return
+
+        requested_fields = [dict(
+            block_type=block_type,
+            field_name=field_name,
+        ) for block_type, field_name in settings.FEATURES.get('COURSE_BLOCKS_API_EXTRA_FIELDS')]
+
+        for block_key in block_structure.topological_traversal():
+            for field in requested_fields:
+                block_type_regex = re.compile(block_structure.get_xblock_field(block_key, 'category'))
+                if block_type_regex.match(field['block_type']) is not None:
```

Comment:
Oh my god. :flushed: You're completely right, I need to compile the field's block type and match it against the actual blog category. I had that in mind, but I often end up writing something different that what I am thinking :sweat: 

I considered regex for matching multiple categories, such as `course|problem`. Do you think that would be useful? Please let me know either way so that I fix the regex or completely remove it. :+1:

Reply:
People can always add multiple tuples if they need multiple block types, e.g. `[('video', 'duration'), ('vimeo_video', 'duration')]` instead of `[('(video|vimeo_video)', 'duration')]`. So maybe just keep it simple and no regex.

But I do think "for all blocks" is useful (we have a use case in mind actually) so supporting `[('*', 'weight'), ]` to mean "allow the `weight` field for all block types" would be useful.

Reply:
Alright, sounds good. I'll go with the solution you suggested in your first comment :+1: